lighthouse-cli/test/smokehouse/firehouse.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Smoke test runner for non-Node environments. Supports skipping and modifiying
+ * expectations to match the environment.
+ */
+
+/* eslint-disable no-console */
+
+const log = require('lighthouse-logger');
+const smokeTests = require('./smoke-test-dfns.js');
+const {collateResults, report} = require('./smokehouse-report.js');
+
+/**
+ * @param {Smokehouse.FirehouseOptions} options
+ */
+async function runSmokes(options) {
+  const {runLighthouse, filter, skip, modify} = options;
+
+  let passingCount = 0;
+  let failingCount = 0;
+
+  for (const test of smokeTests) {
+    for (const expected of test.expectations) {
+      if (filter && !expected.lhr.requestedUrl.match(filter)) {
```

Comment:
given `skip` and `modify` behavior I'd expect filter to be a function, maybe rename to `urlFilterRegex` or something?

Reply:
i like `urlFilter`. I don't like Hungarian typing :p

Reply:
hey don't hate on us hungarians, we're a smart bunch :P

Reply:
`urlFilterPattern`?

Reply:
to be clear it's the fact that we're mixing things of very different types with similar names that's the problem so some sort of type on name would exactly clear that up 😆

Reply:
done